app/models/manageiq/providers/google/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,335 @@
+
+      yield persister_instance, parent_image_uid, flavor, instance
+    end
+  end
+
+  # Adding global key-pairs (needed when instances count == 0)
+  def key_pairs
+    project_key_pairs.each do |ssh_key|
+      persister.key_pairs.build(
+        :name        => ssh_key[:name],
+        :fingerprint => ssh_key[:fingerprint]
+      )
+    end
+  end
+
+  def instance_key_pairs(persister_instance, instance)
+    # Add project common ssh-keys with keys specific to this instance
+    instance_ssh_keys = project_key_pairs | parse_compute_metadata_ssh_keys(instance.metadata)
+
+    instance_ssh_keys.each do |ssh_key|
+      key_pair = persister.key_pairs.build(
+        :name        => ssh_key[:name],       # manager_ref
+        :fingerprint => ssh_key[:fingerprint] # manager_ref
+      )
+
+      key_pair.vms ||= []
+      key_pair.vms << persister_instance
+    end
+  end
+
+  def instance_hardware(persister_instance, series)
+    persister_hardware = persister.hardwares.build(
+      :vm_or_template  => persister_instance, # manager_ref
+      :cpu_total_cores => series[:cpus],
+      :memory_mb       => series[:memory] / 1.megabyte
+    )
+    yield persister_hardware
+  end
+
+  def instance_advanced_settings(persister_instance, instance)
+    persister.vms_and_templates_advanced_settings.build(
+      :resource     => persister_instance, # manager_ref
+      :name         => "preemptible?",
+      :display_name => N_("Is VM Preemptible"),
+      :description  => N_("Whether or not the VM is 'preemptible'. See"\
+                               " https://cloud.google.com/compute/docs/instances/preemptible for more details."),
+      :value        => instance.scheduling[:preemptible].to_s,
+      :read_only    => true
+    )
+  end
+
+  # @param persister_template [InventoryObject]
+  def image_os(persister_template, image)
+    persister.operating_systems.build(
+      :vm_or_template => persister_template, # manager_ref
+      :product_name   => get_os_product_name(image)
+    )
+  end
+
+  # Operating system for Vm (i.e. instance)
+  # It's name is taken from parent image's OS
+  # Note: Not added when parent image not found
+  #
+  # @param persister_vm [InventoryObject] -> IC <ManageIQ::Providers::Google::CloudManager::Vm>
+  # @param parent_image_uid [String] UID of vm's template
+  def instance_os(persister_vm, parent_image_uid)
+    persister.operating_systems.build(
+      :vm_or_template => persister_vm,
+      :product_name   => persister.operating_systems.lazy_find(
+        persister.miq_templates.lazy_find(parent_image_uid),
+        :key => :product_name
+      )
+    )
+  end
+
+  def hardware_disks(persister_hardware, instance)
+    instance.disks.each do |attached_disk|
+      # TODO(mslemr): can't be better solution?
+      cloud_volume_ems_ref = @cloud_volume_url_to_id[attached_disk[:source]]
+      persister_cloud_volume = persister.cloud_volumes.find(cloud_volume_ems_ref)
+      next if persister_cloud_volume.nil?
```

Comment:
@Ladas Also don' know how to remove this `find`. If I replace it with lazy_find, then it will build 'disk' everytime, which is not equivalent.

Reply:
hm, so this differs to other providers, there we build all disks. Basically if it doesn't have a volume, it's a local disk and we do store those in other providers.

Again, lets mark this for later, it will need an exception in specs (since graph refresh will have a different result)